canary/generator/download/download_whole.py

```diff
@@ -0,0 +1,38 @@
+from collections import defaultdict
+import json
+from json import JSONDecodeError
+from ast import literal_eval
+import sys
+import os
+
+import requests
+
+from .query import query
+
+if __name__ == '__main__':
+
+    dates_versions = requests.get(
+        'https://aggregates.telemetry.mozilla.org/aggregates_by/build_id/channels/nightly/dates/').text
+    dates_versions = literal_eval(dates_versions)
+    dates_for_each = defaultdict(list)
+    for item in dates_versions:
+        dates_for_each[item['version']].append(item['date'])
```

Comment:
It's not immediately clear to me what format this has without reading the code in more detail.
Can we name `dates_for_each` more clearly, e.g. something like `version_to_dates`, `dates_for_versions`, ...?
Also, maybe add a comment about what data it holds would help reading (e.g. `# Dict of versions to list of dates.`).